ui/FilesPanel.go

```diff
@@ -90,33 +90,58 @@ func (this *filesPanel) createRefreshButton() *gtk.Button {
 
 func (this *filesPanel) createBackButton() *gtk.Button {
 	image := utils.MustImageFromFileWithSize("back.svg", this.Scaled(40), this.Scaled(40))
-	return utils.MustButton(image, func() {
-		if this.locationHistory.Length() < 1 {
-			this.UI.GoToPreviousPanel()
-		} else if this.locationHistory.IsRoot() {
-			this.locationHistory.GoBack()
-			this.doLoadFiles()
-		} else {
-			this.locationHistory.GoBack()
-			this.doLoadFiles()
-		}
-	})
+	return utils.MustButton(image, this.goBack)
 }
 
 func (this *filesPanel) doLoadFiles() {
 	utils.EmptyTheContainer(&this.listBox.Container)
-
-	if this.displayRootLocations() {
-		this.addRootLocations()
-	} else {
+	if this.isRoot() {
+		if this.refreshSD() {
```

Comment:
`if this.refreshSD()` doesn't read well - it's essentially "if command" vs "if query" (or maybe it's just that I've been working with REST APIs and SQL CRUD for way too many years lol...)

The implementation of refreshSD() is essentially, "refresh the SD list and check if it is ready".  I'd like `refreshSD()` to be renamed to, `sdIsReady()`.  This will read better when grokking the code.

Reply:
Fair. I called it this because the endpoint in OctoPrint is `refresh`, but you are correct, it is doing more than one thing, so the read reading of the action should reflect that.